### PR TITLE
Mat 5085 resource id required

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-5085](https://jira.cms.gov/browse/MAT-5085)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java [hunk 1]
```diff
@@ -59,24 +61,34 @@ public OperationOutcome validateBundleResourcesProfiles(IBaseBundle bundleResour
...
             operationOutcome,
             OperationOutcome.IssueSeverity.ERROR.toCode(),
-            formatUniqueIdViolationMessage(resourceId),
+            "All resources must have an Id",
             null,
             OperationOutcome.IssueType.INVALID.toCode());
-        duplicateIds.add(resourceId);
       } else {
-        existingIds.add(resourceId);
+        if (existingIds.contains(resourceId) && !duplicateIds.contains(resourceId)) {
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java:80]: Can you explain why we need duplicate Ids HashSet?

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java:80]: @RohitKandimalla I think it was to prevent duplicate Id for resources.

Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java:80]: existingIds.contains(resourceId) Isn't this check dong the same thing ?

Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ResourceValidationService.java:80]: I guess I just couldn't understand the difference between "existingIds" and "duplicateIds"

**Review by RohitKandimalla — APPROVED** [review]
